docs/USAGE-GUIDE.md

```diff
@@ -1401,6 +1401,22 @@ ssize_t s2n_client_hello_get_extension_by_id(struct s2n_client_hello *ch, s2n_tl
 **s2n_client_hello_get_extension_length** returns the number of bytes the given extension type takes on the ClientHello message received by the server; it can be used to allocate the **out** buffer.
 **s2n_client_hello_get_extension_by_id** copies into the **out** buffer **max_length** bytes of a given extension type on the ClientHello and returns the number of copied bytes.
 
+### s2n\_client\_hello\_get\_session\_id
+
+```c
+ssize_t s2n_client_hello_get_session_id_length(struct s2n_client_hello *ch);
+ssize_t s2n_client_hello_get_session_id(struct s2n_client_hello *ch, uint8_t *out, uint32_t max_length);
+```
+
+- **ch** The s2n_client_hello on the s2n_connection. The handle can be obtained using **s2n_connection_get_client_hello**.
+- **out** Pointer to a buffer into which the session id bytes should be copied.
+- **max_length** Max number of bytes to copy into the **out** buffer.
+
+**s2n_client_hello_get_session_id_length** returns the number of bytes the session id takes on the ClientHello message received by the server; it can be used to allocate the **out** buffer.
```

Comment:
```suggestion
**s2n_client_hello_get_session_id_length** returns the length of the ClientHello session id in bytes. It can be used to allocate the **out** buffer.
```